app/src/androidTest/java/ch/epfl/sdp/map/fragment/MapActivityTest.java

```diff
@@ -43,19 +43,28 @@
 
 public class MapActivityTest {
 
+    @Rule
+    public final ActivityTestRule<MapActivity> activityRule = new ActivityTestRule<>(MapActivity.class);
+    @Rule
+    public GrantPermissionRule locationPermissionRule = GrantPermissionRule.grant(android.Manifest.permission.ACCESS_FINE_LOCATION);
     private MapFragment mapFragment;
     private AtomicInteger sentinel;
     private MockLocationBroker mockLocationBroker;
     private boolean pathCoordIsEmpty;
     private boolean infectedCoordIsEmpty;
 
-    @Rule
-    public GrantPermissionRule locationPermissionRule =
-            GrantPermissionRule.grant(android.Manifest.permission.ACCESS_FINE_LOCATION);
+    @BeforeClass
+    public static void preSetup() {
+        AccountFragment.IN_TEST = true;
+        MapFragment.TESTING_MODE = true;
+    }
 
-    @Rule
-    public final ActivityTestRule<MapActivity> activityRule =
-            new ActivityTestRule<>(MapActivity.class);
+    @AfterClass
+    public static void postClean() {
+        AccountFragment.IN_TEST = false;
```

Comment:
We should create a single global variable that can take three values from an ENUM: APP, TEST, DEMO

Reply:
Completely agree. I think it's worth to dedicate a branch to better handle those different scenarios